kitchen-sink-example.yaml

```diff
@@ -26,7 +71,44 @@ logger_provider:
     attribute_count_limit: 128
 
 # Configure meter provider.
-meter_provider: {}
+meter_provider:
+  metric_readers:
+    # Add a pull-based metric reader.
+    - type: pull
+      args:
+        exporter:
+          <<: *prometheus-exporter
+    # Add a periodic metric reader.
+    #
+    # Environment variable: OTEL_METRICS_EXPORT_*, OTEL_METRICS_EXPORTER
+    - type: periodic
+      args:
```

Comment:
> We could have metric_readers be an object instead of array, with keys which refer to the type, and values whose schema is enforced based on the key

How would this work in the case where a user may want to configure multiple periodic exporters?

Reply:
Same way we were planning on supporting multiple exporters where the keys identify the exporter:
```
metric_readers:
  periodic/foo: ...
  periodic/bar: ...
```
I don't love it... Better if we can find a way to do:
> Maybe the best we can do is find a way to conditionally enforce a schema on args based on the value of type
But if you agree, that can be in a subsequent PR.

Reply:
Oh right.... yes that's fine

Reply:
I've updated the schema to follow @jack-berg's suggestion